ci/check-protobuf-stability.sh

```diff
@@ -35,42 +51,76 @@ function check_non_lf_protos() {
   readonly BUF_IMAGE_TMPDIR="$(mktemp -d)"
   trap 'rm -rf ${BUF_IMAGE_TMPDIR}' EXIT
 
-  # The `SYSTEM_PULLREQUEST_TARGETBRANCH` environment variable is defined by
-  # Azure Pipelines; in order to run this script locally, define it beforehand
-  # as the branch being targeted. For example:
-  #
-  #   SYSTEM_PULLREQUEST_TARGETBRANCH=main bash -x ci/check-protobuf-stability.sh
-  #
-  TARGET="${SYSTEM_PULLREQUEST_TARGETBRANCH:-main}"
-  echo "The target branch is '${TARGET}'."
+  echo "Checking protobufs against git target '${BUF_GIT_TARGET_TO_CHECK}'"
+  for buf_module in "${BUF_MODULES_AGAINST_STABLE[@]}"; do
+    # Starting with version 1.17 we split the default `buf.yaml` file into multiple config files
+    # This in turns requires that we pass the `--against-config` flag for any check that is run on versions > 1.17
+    if [[ $BUF_CONFIG_UPDATED = true ]]
+    then
+      buf breaking --config "${buf_module}" --against "$BUF_GIT_TARGET_TO_CHECK" --against-config "${buf_module}"
+    else
+      buf breaking --config "${buf_module}" --against "$BUF_GIT_TARGET_TO_CHECK"
+    fi
+  done
 
-  # For `main` and PRs targeting `main`, we simply check against the most recent
-  # stable tag.
-  #
-  # For PRs targeting release branches, we should really check against
-  # all the most recent stable tags reachable from either the current branch or
-  # from previous release branches (say, against both `1.17.1` and `1.16.2`
-  # created after the `release/1.17.x` branch).
-  # Instead, we approximate by checking only against the most recent stable tag
-  # reachable from the current branch, under the assumption that if a lesser
-  # release branch contains a protobuf change, then it will also be present in
-  # higher ones either through a shared commit or a back-port from `main`.
+}
+
+case "${1:---stable}" in
+-h | --help)
+  cat <<USAGE
+Usage: ./check-protobuf-stability.sh [options]
+
+Options:
+  -h, --help: shows this help
+  --stable:     check against the latest stable tag (default)
+  --target:     check against the tip of the target branch
+USAGE
+  exit
+  ;;
+--stable)
+  # Check against the most recent stable tag.
   #
-  # Finally, this check does not need to run on release branch commits because
+  # This check does not need to run on release branch commits because
   # they are built sequentially, so no conflicts are possible and the per-PR
   # check is enough.
+  readonly RELEASE_BRANCH_REGEX="^release/.*"
   GIT_TAG_SCOPE=""
+  # For PRs targeting release branches, we should really check against
+  # all the most recent stable tags reachable from either the current branch or
+  # from previous release branches (say, against both 1.17.1 and 1.16.2
+  # created after the release/1.17.x branch).
+  #
```

Comment:
If you decide to leave this, then the next paragraph's start should be adjusted to something like "We rather check".